edocumentation-service/src/main/java/org/dcsa/edocumentation/service/mapping/BookingSummaryMapper.java

```diff
@@ -0,0 +1,36 @@
+package org.dcsa.edocumentation.service.mapping;
+
+import org.dcsa.edocumentation.domain.persistence.entity.Booking;
+import org.dcsa.edocumentation.transferobjects.BookingSummaryTO;
+import org.dcsa.edocumentation.transferobjects.enums.DCSATransportType;
+import org.dcsa.skernel.errors.exceptions.ConcreteRequestErrorMessageException;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+
+@Mapper(componentModel = "spring")
+public interface BookingSummaryMapper {
+
+  @Mapping(source = "booking.vessel.vesselIMONumber", target = "vesselIMONumber")
+  @Mapping(source = "updatedDateTime", target = "bookingRequestUpdatedDateTime")
+  @Mapping(source = "bookingRequestDateTime", target = "bookingRequestCreatedDateTime")
+  @Mapping(source = "preCarriageModeOfTransportCode", target = "preCarriageModeOfTransportCode", qualifiedByName = "toDCSATransportType")
+  BookingSummaryTO BookingToBookingSummary(Booking booking);
+
+  @Named("toDCSATransportType")
+  static DCSATransportType toDcsaTransportType(String modeOfTransportCode) {
+
+    if (modeOfTransportCode == null) {
+      return null;
+    }
+
+    return switch (modeOfTransportCode) {
+      case "1" -> DCSATransportType.VESSEL;
+      case "2" -> DCSATransportType.RAIL;
+      case "3" -> DCSATransportType.TRUCK;
+      case "8" -> DCSATransportType.BARGE;
+      default -> throw ConcreteRequestErrorMessageException.internalServerError("Could not convert TransportTypeCode to DCSA TransportType");
+
+    };
```

Comment:
These values are in the database, so we could just fetch them as a separate entity (which makes it easier to support new values as they are added and we can use our standard processes for the value).

Reply:
(fetch them via a JOIN)

Reply:
fixed